### PR TITLE
GLAD/NBR/COPING PGCED freeze 3

### PR DESCRIPTION
**Pull request by Helena-D**

### CODE REVIEW

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 1]
```diff
@@ -0,0 +1,4291 @@
...
+  freq(health.IBD)
+
+dat_nbr_health.id %>%
+  freq(health.IBS)
+```
+
+## Next, need to consider AN screener question
+### Read in data: AN screener
+```{r NBR data AN screener}
+dat_nbr_AN_screener <- readRDS(file = "/Users/helenadavies/King's College London/MT-TNG BioResource EDIT - ilovedata - ilovedata/data_raw/latest_freeze/coping_nbr/an_scr_coping_nbr.rds")
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: path

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 2]
```diff
@@ -0,0 +1,4291 @@
...
+
+# Inspect dimensions
+dat_nbr_AN_screener %>%
+  dim()
+
+# Inspect colnames
+dat_nbr_AN_screener %>%
+  colnames()
+
+dat_nbr_AN_screener %>%
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: dim and nrow is a duplicate

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 3]
```diff
@@ -0,0 +1,4291 @@
...
+         )
+
+# Check
+dat_nbr_AN_screener.id %>%
+  dim()
+
+dat_nbr_AN_screener %>%
+  colnames()
+
+dat_nbr_AN_screener %>%
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: dim and nrow duplicate

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 4]
```diff
@@ -0,0 +1,4291 @@
...
+)
+
+# Check
+dat_nbr_merged %>%
+  colnames()
+
+dat_nbr_merged %>%
+  nrow()
+```
+
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: I don't understand why you don't define them as the inverse of your cases?
meaning not having AN, ANR, ANBP, BN, or BED or any other ED from the MHD?

Comment (topherhuebel) [reply GLAD_NBR_PGCED3_cases.Rmd]: The controls should be the same controls that we are using for all analysis?!

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 5]
```diff
@@ -0,0 +1,4291 @@
...
+## Control algorithm
+1. Any individual who does not have a history of binge eating (narrow or broad definition) AND does not have a history of any eating disorder.
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: Binge eating, but then you say no_AN?
Why is no_AN not just
not AN, ANR, ANBP, BN, BED?

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 6]
```diff
@@ -0,0 +1,4291 @@
...
+              (aan.considered_lost_weight_period_cop == "No" | # Did not pass AAN screener
+                ed.DSM5_AN_atypical_low_BMI_binary_numeric_cop == 0) ~1  # OR did pass AAN screener but do not meet diagnostic criteria for anorexia nervosa via the AAN module
+           )
+         )
+
+
+dat_nbr_merged %>%
+  freq(no_AN)
+
+# No atypical anorexia
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: They should just be your 0 of AAN when using the broadest definition of AAN?!

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 7]
```diff
@@ -0,0 +1,4291 @@
...
+             )
+         )
+
+dat_nbr_merged %>%
+  freq(no_BE)
+
+freq(dat_nbr_merged$mhd.none_eds_cop_numeric)
+
+
+# No history of other EDs (self-report)
```
Comment (topherhuebel) [inline GLAD_NBR_PGCED3_cases.Rmd]: Again, doesn't really make sense to define it like this or am I missing something?

**Review by topherhuebel — CHANGES_REQUESTED** [review]
@Helena-D
The script is looking good to me. Just a lot of tiny things. 
I recommend splitting the monster in several subscripts that we can run subsequently with a wrapper script.
If you split the script up, I’m happy to help with re-writing in tidyverse. We could do this script by script, because we need the variables also for EDGI analyses.
Please also use a credentials file for the file paths.
I’m having 1000 lines left, but I think if you could clean up the script and split it, it will make more sense than me saying the same thing over and over again

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 8]
```diff
@@ -0,0 +1,4291 @@
...
+        #and these episodes occurred on average at least once a week for at least three months (= DSM-5 frequency and duration criteria);
+            (
+              COPING_BE_duration_months >= 3 &
+                COPING_binge_freq_per_wk >= 1 
+            )
+          ) |
+    
+          #3. Has a self-reported diagnosis of BN or BED, with specification that this has been confirmed by a healthcare professional (COPING);
+          (
+           mhd.bn_cop_numeric ==1 |
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: + add to annotation (that ANB cases would be captured by lifetime bigne eating)

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 9]
```diff
@@ -0,0 +1,4291 @@
...
+```
+
+ANR- Identify those who were classed as cases in COPING GLAD only, GLAD only, or both
+```{r ANR identify cases GLAD, COPING GLAD, NBR}
+dat_with_selfreport <- dat_with_selfreport %>%
+  mutate(
+    cases_ANR =
+           case_when(
+             ANR_COPING == 1 &
+               ANR_GLAD == 1 ~ "Both",
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: EDGI = no genotype data

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 10]
```diff
@@ -0,0 +1,4291 @@
...
+```
+
+# ANOREXIA PHENOTYPES
+## Anorexia nervosa binge-eating/purging subtype: 
+
+Any individual who meets the following criteria will be coded as “ANBP = 1”:
+
+    • Has received a clinical diagnosis of anorexia nervosa binge-eating/purging subtype via hospital or register records or structured clinical interviews, or online questionnaires based on standardized criteria (DSM III-R, DSM-IV, DSM 5, ICD-9, or ICD-10); 
+
+    • Has a self-reported diagnosis of anorexia nervosa binge-eating/purging (specification that this has been confirmed by a healthcare professional is not required for this definition)
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: ANB is covered by 'binge eating'. we have 5 GWASs: ANBP, ANR, AN, narrow BE, broad BE

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 11]
```diff
@@ -0,0 +1,4291 @@
...
+              be.loss_control == "Very" |
+              be.loss_control == "Extremely"
+          ) 
+          ) |
+
+           # 3. Has a self-reported diagnosis of BN or BED (specification that this has been confirmed by a healthcare professional is not required for this definition);
+       
+             # MHD:
+              (
+           mhd.bn_numeric == 1 |
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: covered by binge eating

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 12]
```diff
@@ -0,0 +1,4291 @@
...
+## BROADLY DEFINED binge eating
+
+All individuals who meet any of the following criteria and will be coded as “BE_broad = 1”:
+
+1.	Eating an unusually large amount of food in an unusually short period of time with loss of control when doing so. Frequency and duration criteria are not required.  
+
+2.	If binge eating (BE) is assessed in a single question like: “Have you ever experienced binge eating?” and is endorsed, they will be considered to have BE and be included in the primary BE GWAS. If the question combines several descriptors such as “Have you ever experienced binge eating OR other terms like psychological overeating, compulsive eating, emotional eating, etc.”), it should not be used for the primary BE GWAS but can be used for PRS analyses.
+
+3.	Has a self-reported diagnosis of BN or BED (specification that this has been confirmed by a healthcare professional is not required for this definition);
+  a.	If ANBP or OSFED/EDNOS is diagnosed, these individuals will only be included if they explicitly state ‘subthreshold BN’ or ‘subthreshold BED’.
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: covered by binge eating

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 13]
```diff
@@ -0,0 +1,4291 @@
...
+```{r ANR algorithm PGC GLAD}
+dat_with_selfreport <- dat_with_selfreport %>%
+    mutate(
+      ANR_GLAD =
+        if_else(
+          # GLAD
+        (  
+          # Has received a clinical diagnosis of anorexia nervosa restricting via hospital or register records or structured clinical interviews, or online questionnaires based on standardized criteria (DSM III-R, DSM-IV, DSM 5, ICD-9, or ICD-10); 
+          (
+            ed.DSM5_AN_restricting_bmi_20_binary_numeric == 1
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: to check!

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 14]
```diff
@@ -0,0 +1,4291 @@
...
+
+# Get rid of implausible values, i.e., values below 0
+dat_with_selfreport$aan.weight_prior_kg_total_cop[dat_with_selfreport$aan.weight_prior_kg_total_cop < 20 ] <- NA_real_
+dat_with_selfreport$aan.weight_prior_kg_total_cop[dat_with_selfreport$aan.weight_prior_kg_total_cop > 635 ] <- NA_real_ #Heaviest person in world is 635kg
+
+# Check again
+freq(dat_with_selfreport$aan.weight_prior_kg_total_cop)
+```
+
+Work out weight at period of significant weight loss (COPING GLAD & NBR)
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: this is weight AFTER weight loss. aan.weight_prior_kg_total_cop is weight BEFORE

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 15]
```diff
@@ -0,0 +1,4291 @@
...
+    descr(
+      AAN_BMI_cop
+    )
+```
+
+## Work out AAN BMI (GLAD)
+Amount of weight lost (GLAD)
+```{r AAN amount of weight loss GLAD}
+# Rename / convert to numeric
+dat_with_selfreport$aan.weight_lost_lbs <- dat_with_selfreport$aan.stone_pounds_lose_note_numeric
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: was already numeric

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 16]
```diff
@@ -0,0 +1,4291 @@
...
+
+# Identify number of outliers
+  length(
+    which(
+      dat_with_selfreport$AAN_BMI > bmi_upper_limit |
+        dat_with_selfreport$AAN_BMI < bmi_lower_limit
+    )
+  )
+  
+# Remove age outliers
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: *bmi outliers!

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 17]
```diff
@@ -0,0 +1,4291 @@
...
+All of the criteria for anorexia nervosa are met, except that despite significant weight loss, the individual’s weight is within or above the normal range (BMI of 20 or above)
+```{r AAN GLAD algorithm low BMI}
+#Atypical anorexia nervosa numeric variable
+dat_with_selfreport$ed.DSM5_AN_atypical_low_BMI_binary_numeric <- 
+  with(dat_with_selfreport,
+  dplyr::if_else(
+    
+# BMI less than 18.5
+  
+        (AAN_BMI <= 18.55 &
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: yes exactly - these are counted as AN cases. will rename 'AN_from_AAN_module'

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 18]
```diff
@@ -0,0 +1,4291 @@
...
+dat_with_selfreport$ed.DSM5_AN_atypical_low_BMI_binary <-
+  recode_factor(dat_with_selfreport$ed.DSM5_AN_atypical_low_BMI_binary_numeric,
+                "0" = "No DSM-5 atypical AN BMI less than 18.5",
+                "1" = "DSM-5 Atypical AN BMI less than 18.5")
+
+# Summary
+freq(dat_with_selfreport$ed.DSM5_AN_atypical_low_BMI_binary)
+```
+
+GLAD AAN Low BMI Count cases
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: yes - convoluted i know......

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 19]
```diff
@@ -0,0 +1,4291 @@
...
+
+# Double check which cohort they are from
+dat_with_selfreport %>%
+  filter(AAN_LOWBMI_GLAD == 1)%>%
+  group_by(sample) %>%
+  count()
+```
+
+### BMI of 18.5 or less [COPING GLAD and NBR]
+All of the criteria for anorexia nervosa are met, except that despite significant weight loss, the individual’s weight is within or above the normal range (BMI of 20 or above)
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: need to change to AN definition

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 20]
```diff
@@ -0,0 +1,4291 @@
...
+  dplyr::if_else(
+    
+#BMI less than 18.5
+  
+       (AAN_BMI_cop <= 18.55 &
+        
+         # Afraid of gaining weight 
+        (aan.1.gain_weight_afraid_fat_cop == "Somewhat afraid" | 
+         aan.1.gain_weight_afraid_fat_cop == "Very afraid" |
+         aan.1.gain_weight_afraid_fat_cop == "Extremely afraid") &
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: this is AN (no subtype) and we have not changed the BMI threshold

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 21]
```diff
@@ -0,0 +1,4291 @@
...
+  mutate(AN_CASE_GWAS =
+           case_when(AN_COPING == 1 |
+                     AN_GLAD == 1 |
+                     ANBP_GLAD == 1 |
+                     ANBP_COPING == 1 |
+                     ANR_GLAD == 1 |
+                     ANR_COPING == 1 |
+                     AAN_LOWBMI_COPING == 1 |
+                     AAN_LOWBMI_GLAD == 1 ~ 1
+
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: i have done them separately

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 22]
```diff
@@ -0,0 +1,4291 @@
...
+## 'Anorexia nervosa' with BMI over 18.55 [COPING]
+```{r AN COPING algorithm high BMI}
+dat_with_selfreport$ed.DSM5_AN_high_BMI_binary_numeric_cop <- 
+  with(dat_with_selfreport,
+  dplyr::if_else(
+#A. Restriction of energy intake relative to requirements,leading to a significantly low body weight in the context of age, sex, developmental trajectory, and physical health. Significantly low weight is defined as a weight that is less than minimally normal or, for children and adolescents, less than that minimally expected.
+     (
+     (
+    an.bmi_lowest_cop > 18.55  # Lowest BMI as reported in ED100K optional questionnaire AN module, greater than 18.55
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: our definition of AAN is BMI above 18.5

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 23]
```diff
@@ -0,0 +1,4291 @@
...
+dat_nbr_health.id %>%
+  dim()
+```
+
+Frequency of IBS and/or IBD
+```{r Freq IBD }
+dat_nbr_health.id %>%
+  freq(health.IBD)
+
+dat_nbr_health.id %>%
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: one is IBD, the other is IBS

File: GLAD_NBR_PGCED3_cases.Rmd (hunk 5 above)
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: There may be people who self-report BN/BED but do not endorse binge eating in the ED100K - will do a cross table to check this

File: GLAD_NBR_PGCED3_cases.Rmd (hunk 6 above)
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: need to include the screener - people who say 'No' to screener need to be 0, the algorithm does not compute

File: GLAD_NBR_PGCED3_cases.Rmd (hunk 7 above)
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: People who suspected but never been diagnosed with an eating disorder - ask about this agian!!!!!!

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 24]
```diff
@@ -0,0 +1,4291 @@
...
+          # with loss of control when doing so...
+          (
+              be.loss_control == "Slightly" | 
+              be.loss_control == "Somewhat" |
+              be.loss_control == "Very" |
+              be.loss_control == "Extremely"
+          ) &
+        # and these episodes occurred on average at least once a week for at least three months (= DSM-5 frequency and duration criteria);
+            (
+              GLAD_BE_duration_months >= 3 &
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: the GLAD variables have "GLAD_" at the beginning whilst the GLAD COPING ones are left as they are, whilst the GLAD COPING have "COPING"

File: GLAD_NBR_PGCED3_cases.Rmd [hunk 25]
```diff
@@ -0,0 +1,4291 @@
...
+                GLAD_BE_freq_per_week >= 1 
+            )
+          ) |
+
+         #   2. No clinical records in GLAD (yet)
+          
+          # 3. Has a self-reported diagnosis of BN or BED, with specification that this has been confirmed by a healthcare professional (GLAD);
+          (
```
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: that is captured by binge eating

File: GLAD_NBR_PGCED3_cases.Rmd (hunk 8 above)
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: done!

File: GLAD_NBR_PGCED3_cases.Rmd (hunk 19 above)
[the remark this answers is not in the document]
Comment (Helena-D) [reply GLAD_NBR_PGCED3_cases.Rmd]: done (removed it)

**Comment by Helena-D** [comment]
adding all files back (also hcnages to GLAD/NBR script following Topher's review)

**Pull request merged**